internal/stack/_static/serverless-logstash.conf.tmpl

```diff
@@ -0,0 +1,33 @@
+input {
+  elastic_agent {
+    port => 5044
+    ssl_enabled => true
+    ssl_certificate_authorities => ["/usr/share/logstash/config/certs/ca-cert.pem"]
+    ssl_certificate => "/usr/share/logstash/config/certs/cert.pem"
+    ssl_key => "/usr/share/logstash/config/certs/logstash.pkcs8.key"
+    ssl_client_authentication => "required"
+  }
+}
+
+
+filter {
+  elastic_integration {
+    remove_field => ['@version']
+    hosts => ["{{ fact "elasticsearch_host" }}"]
+    username => '{{ fact "username" }}'
+    password => '{{ fact "password" }}'
+    ssl_enabled => true
+    ssl_verification_mode => "full"
+  }
+}
+
+
+output {
+  elasticsearch {
+    hosts => ["{{ fact "elasticsearch_host" }}"]
+    user => '{{ fact "username" }}'
+    password => '{{ fact "password" }}'
+    ssl_enabled => true
+    data_stream => "true"
```

Comment:
This is realy great change and thanks alot @bhapas !
I have also went through the changes and LGTM.

I know we are touching serverless point here but since we are improving interfacing with Logstash on high level, I would love if we can include is, similar to this config, remove `document_id` from [on-prems config](https://github.com/elastic/elastic-package/blob/main/internal/stack/_static/logstash.conf.tmpl#L32). Setting document id is not always a right way since ES generates document ID and especially it becomes issue when we use integrations which are based on time series.

- `/internal/stack/_static/logstash.conf.tmpl`

```diff
-    document_id => "%{[@metadata][_ingest_document][id]}"
```

OR

If it is hard to change, the alternative solution would be making a possibility to change the config file. Currently, it is a read-only. We also need to add an auto reload feature which Logstash can automatically detect and reload the pipeline. This will help us to prevent manual docker restart (with docker container restart we loose JVM truststore auto configs, connections may not be established between containers).
 
- `internal/stack/_static/docker-compose-stack.yml.tmpl` file change

```diff
       retries: 5
-    command: bash -c 'if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf'
+    command: bash -c 'if [[ ! $(bin/logstash-plugin list) == *"logstash-filter-elastic_integration"* ]]; then echo "Missing plugin logstash-filter-elastic_integration, installing now" && bin/logstash-plugin install logstash-filter-elastic_integration; fi && bin/logstash -f /usr/share/logstash/pipeline/logstash.conf --config.reload.automatic'
     volumes:
       - "../certs/logstash:/usr/share/logstash/config/certs"
       - "../certs/elasticsearch/cert.pem:/usr/share/logstash/config/certs/elasticsearch.pem"
-      - "./logstash.conf:/usr/share/logstash/pipeline/logstash.conf:ro"
+      #- "./logstash.conf:/usr/share/logstash/pipeline/logstash.conf:ro"
     ports:

```

Reply:
Hm, as I have tested if we restart the container we no longer can connect to ES, interesting...
```
[2024-02-07T02:05:04,766][WARN ][logstash.outputs.elasticsearch][main] Failed to perform request {:message=>"Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty", :exception=>Manticore::UnknownException, :cause=>#<Java::JavaxNetSsl::SSLException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty>}
[2024-02-07T02:05:04,767][WARN ][logstash.outputs.elasticsearch][main] Attempted to resurrect connection to dead ES instance, but got an error {:url=>"https://elastic:xxxxxx@elasticsearch:9200/", :exception=>LogStash::Outputs::ElasticSearch::HttpClient::Pool::HostUnreachableError, :message=>"Elasticsearch Unreachable: [https://elasticsearch:9200/][Manticore::UnknownException] Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty"}
```

Reply:
@mashhurs I will take up the on-prem elastic stack with logstash changes as part of #1658